File: 03-poo/03-constructores-e-instanciacion/scala/soluciones/RetoProducto.scala
class ProductoReto(val nombre: String, val precio: Double):
  require(precio >= 0, "El precio no puede ser negativo")
  def ficha: String = s"$nombre - S/ $precio"

@main def retoProducto(): Unit = println(new ProductoReto("Teclado", 80).ficha)

File: 03-poo/03-constructores-e-instanciacion/java/soluciones/RetoProducto.java
public class RetoProducto {
    static class Producto {
        private final String nombre;
        private final double precio;

        Producto(String nombre, double precio) {
            if (precio < 0) throw new IllegalArgumentException("El precio no puede ser negativo");
            this.nombre = nombre;
            this.precio = precio;
        }
        String ficha() { return nombre + " - S/ " + precio; }
    }
    public static void main(String[] args) {
        System.out.println(new Producto("Teclado", 80).ficha());
    }
}
